knimin/lib/parse.py

```diff
@@ -94,3 +95,84 @@ def parse_qpcr_object(contents):
         result[r, c] = value
 
     return result
+
+
+def _find_section(lines, label):
+    """Find a section, eg [EXCEPTIONS]
+
+    Parameters
+    ----------
+    lines : list of str
+        The data to search over
+    label : str
+        The section header to search for
+
+    Returns
+    -------
+    int
+        The index of the header
+    int
+        The index of the first empty line following the section, or the index
+        position of the end of the file.
+
+    Raises
+    ------
+    ValueError
+        If the section is not found
+    """
+    start = -1
```

Comment:
This is not performance bound code, and the total number of iterations is the same as at most, the code will the entire file only once as the forloops do not overlap. The multiple loops also represent a logical separation, and I'd argue is more in line with KISS, where each forloop statement does one specific thing.